zcommon/src/org/zkoss/lang/Library.java

```diff
@@ -154,13 +150,11 @@ public static int getIntProperty(String key, int defVal) {
 		if (key == null) throw new NullPointerException();
 		if (key.length() == 0) throw new IllegalArgumentException();
 
-		synchronized (_props) {
-			List<String> old = _props.get(key);
-			if (old == null)
-				old = new LinkedList<String>();
-			old.add(value);
-			return _props.put(key, old);
-		}
+		List<String> old = _props.get(key);
+		if (old == null)
+			old = new LinkedList<String>();
+		old.add(value);
```

Comment:
the old(LinkedList) is not thread-safe, multi-thread add might be wrong. You should protect it.

Reply:
ok